de.metas.swat/de.metas.swat.base/src/main/java/org/adempiere/invoice/process/CreateAdjustmentChargeFromInvoice.java

```diff
@@ -57,7 +58,8 @@ protected void prepare()
 			}
 			else if (CreateAdjustmentChargeFromInvoice.PARA_DocSubType.equals(name))
 			{
-				parameterDocSubType = (BigDecimal)parameter;
+				// convert the given value of the parameter to BigDecimal
+				parameterDocSubType = NumberUtils.asBigDecimal(parameter, new BigDecimal(-1));
```

Comment:
Why is `parameterDocType` a number to start with? 
After all, when selecting `select DocSubType from c_doctype` there isn't one numeric value.

It seems to me that the params got mixed up in the application dictionary and then someone (pre-github times, i didn't check in the old hg repo) just changed the field's type. 

Please investigate and fix this further. If the docSubtype parameter isn't actually needed (anymore), then please remove it and clean up the API.

Generally, when encountering such a rather outdated piece of code and it will need to go through IT anyways, then please modernize it and clean it up. 
Here, i think that means:
* see what those AD_Table_ID and Record_ID fields are about and remove them if they are not needed (as it seems to me)
* get rid of the `prepare()`method and use the `@Param` annotation instead.